app.js

```diff
@@ -42,11 +42,44 @@ btns.forEach((button) => {
   button.addEventListener("click", (e) => calculatorDisplay(button.id));
 });
 
+window.addEventListener("keydown", (e) => {
+  if (e.keyCode === 48) {
+    calculatorDisplay("zero");
+  } else if (e.keyCode === 49) {
+    calculatorDisplay("one");
+  } else if (e.keyCode === 50) {
+    calculatorDisplay("two");
+  } else if (e.keyCode === 51) {
+    calculatorDisplay("three");
+  } else if (e.keyCode === 52) {
+    calculatorDisplay("four");
+  } else if (e.keyCode === 53) {
+    calculatorDisplay("five");
+  } else if (e.keyCode === 54) {
+    calculatorDisplay("six");
+  } else if (e.keyCode === 55) {
+    calculatorDisplay("seven");
+  } else if (e.keyCode === 56) {
+    calculatorDisplay("eight");
+  } else if (e.keyCode === 57) {
+    calculatorDisplay("nine");
+  } else if (e.keyCode === 190) {
+    calculatorDisplay("decimal");
+  } else if (e.keyCode === 8) {
+    calculatorDisplay("undo");
+  }
```

Comment:
This code is extremely repetitive: you can again reduce it down by a lot if you use a "lookup table":

```js
const keyCodeToOp = {
  49: "one",
  50: "two",
  51: "three",
  52: "four",
  53: "five",
  54: "six",
  55: "seven",
  56: "eight",
  57: "nine",
  190: "decimal",
  8: "undo",
}
calculatorDisplay(keyCodeToOp(e.keyCode));
```

I would also recommend using even.key instead of event.keyCode as it's much more readable. Also because keyCode doesn't take modifiers into account, and key codes are different for eg the numpad dot and the regular dot. I have no numpad, but I can type a dot, should I be unable to use your calculator with the keyboard?